commands/cli.go

```diff
@@ -63,6 +84,85 @@ func parseDurationParam(paramValue string) (*time.Time, error) {
 	return &excludeAfter, nil
 }
 
+func gcpNuke(c *cli.Context) error {
+	ctx, err := gcp.DefaultContext()
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	logging.Logger.Infof("Using project: %s", ctx.Project)
+
+	excludedRegions := c.StringSlice("exclude-region")
+
+	for _, excludedRegion := range excludedRegions {
+		if !ctx.ContainsRegion(excludedRegion) {
+			return InvalidFlagError{
+				Name:  "exclude-regions",
+				Value: excludedRegion,
+			}
+		}
+	}
+
+	excludeAfter, err := parseDurationParam(c.String("older-than"))
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	logging.Logger.Infoln("Retrieving all active GCP resources")
+
+	resources, err := ctx.GetAllResources(excludedRegions, *excludeAfter)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	if len(resources) == 0 {
+		logging.Logger.Infoln("Nothing to nuke, you're all good!")
+		return nil
+	}
+
+	logging.Logger.Infoln("The following GCP resources are going to be nuked: ")
+
+	for _, resource := range resources {
+		logging.Logger.Infof("* %s: %s Region=%s %s=%s",
+			resource.ResourceName(), resource.Name(), resource.Region(),
+			resource.LocationName(), resource.Location())
+	}
+
+	if !c.Bool("force") {
+		color := color.New(color.FgHiRed, color.Bold)
+		color.Println("\nTHE NEXT STEPS ARE DESTRUCTIVE AND COMPLETELY IRREVERSIBLE, PROCEED WITH CAUTION!!!")
+
+		prompt := "\nAre you sure you want to nuke all listed resources? Enter 'nuke' to confirm: "
+		shellOptions := shell.ShellOptions{Logger: logging.Logger}
+		input, err := shell.PromptUserForInput(prompt, &shellOptions)
+
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+
+		if strings.ToLower(input) == "nuke" {
+			err := ctx.NukeAllResources(resources)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		logging.Logger.Infoln("The --force flag is set, so waiting for 10 seconds before proceeding to nuke everything in your project. If you don't want to proceed, hit CTRL+C now!!")
+		for i := 10; i > 0; i-- {
+			fmt.Printf("%d...", i)
+			time.Sleep(1 * time.Second)
+		}
+
+		fmt.Println()
+		err := ctx.NukeAllResources(resources)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
This if/else prompt code is prob identical to the AWS version, so perhaps extract into a reusable `promptForConfirmationBeforeNuking()` function.